src/client.ts

```diff
@@ -231,6 +233,15 @@ export class SkynetClient {
 
     const auth = config.APIKey ? { username: "", password: config.APIKey } : undefined;
 
+    const onDownloadProgress =
+      config.onDownloadProgress &&
+      /* istanbul ignore next */
+      function (event: ProgressEvent) {
+        const progress = event.loaded / event.total;
```

Comment:
Is `event.total` ever 0? I'm thinking about empty file uploads (of which we have a ton). Should we take that edge into account here?